account_brand/models/res_partner_account_brand.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+
+
+class ResPartnerAccountBrand(models.Model):
+    """This model is meant to be used in case we need to define different
+    receivable/payable accounts for partners"""
+
+    _name = "res.partner.account.brand"
+    _description = "Receivable/Payable Partner Account By Brand"
+
+    partner_id = fields.Many2one(
+        comodel_name="res.partner", string="Partner", required=False
+    )
+    account_id = fields.Many2one(
+        comodel_name="account.account", string="Account", required=True
+    )
+    brand_id = fields.Many2one(
+        comodel_name='res.brand',
+        string='Brand',
+        required=True,
+    )
+    account_type = fields.Selection(
+        string="Type",
+        selection=[("payable", "Payable"), ("receivable", "Receivable")],
+        required=True,
+    )
```

Comment:
Do we want a constraint that checks that `account_id.user_type_id.type` is compatible the same as `account_type`?